From a6fd72cc5a500a2b8f896e48f35b18800860aa16 Mon Sep 17 00:00:00 2001
From: Rameshkumar Sundaram <quic_ramess@quicinc.com>
Date: Mon, 8 Jan 2024 21:10:16 +0530
Subject: [PATCH] wifi: nl80211: allow MBSSID TX VAP bringup without MBSSID IEs

Current implementation of MBSSID configuration parsing mandates
MBSSID IEs for Tx VAP (index 0) however with ML link addition
it is possible that Non-Tx BSS'es can be added later point in
time after TX BSS is brought up.

Hence allow bringup of MBSSID Tx BSS even if no Non-Tx BSS are
present at that time. Later when new Non-TX BSS are added TX BSS
beacon can be updated with MBSSID IEs.

Signed-off-by: Rameshkumar Sundaram <quic_ramess@quicinc.com>
---
 net/wireless/nl80211.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/net/wireless/nl80211.c b/net/wireless/nl80211.c
index 07df662..c6b7756 100644
--- a/net/wireless/nl80211.c
+++ b/net/wireless/nl80211.c
@@ -5978,8 +5978,7 @@ static int nl80211_parse_mbssid_config(struct wiphy *wiphy,
 	}
 
 	config->index = nla_get_u8(tb[NL80211_MBSSID_CONFIG_ATTR_INDEX]);
-	if (config->index >= wiphy->mbssid_max_interfaces ||
-	    (!config->index && !num_elems))
+	if (config->index >= wiphy->mbssid_max_interfaces)
 		return -EINVAL;
 
 	if (tb[NL80211_MBSSID_CONFIG_ATTR_TX_IFINDEX]) {
-- 
2.34.1

